### DISCUSSION TITLE
Feature/Doc Request: Unique Constraint Across Columns

### DISCUSSION THREAD

**Discussion by toadslop** [SeaQL/sea-orm#646] [Ideas]
## Summary

This may either be a documentation request or a feature request.
Most databases support unique contraints across columns such that there can be no record in the table where the combination of values between two columns are identical. So far from the documentation I can't find anything about this feature.

If the feature already exists, I'd like some documentation about it.

If it doesn't exist yet, I'd like it added.

## Motivation

It's a basic functionality of most databases.
If SeaORM doesn't support it, it means that someone would have to write an entire table definition in raw SQL just to use that feature.
It's incredibly inconvenient.

On the other side, if the feature already exists but isn't documented, I think many people will end up wasting many hours experimenting and/or reading through the documentation for a clue that may or may not be there.

**Comment by billy1624** [comment]
Hey @toadslop, good questions! Unique constraint of single column is supported. See https://www.sea-ql.org/SeaORM/docs/generate-entity/entity-structure#additional-properties. However, unique constraint with multiple columns is not supported at the moment.

**Reply by fengbenming** [reply]
Support unique constraint with multiple columns by the placeholder is convient, like gorm. : )

**Comment by tyt2y3** [comment]
First of all, you CAN create unique constraints across multiple columns using SeaQuery.
There is just no way to express this information in SeaORM's Entity.
Anyway, for now, the `unique` or `indexed` properties are just placeholders, they don't affect query generation in any way.

**Comment by hgiesel** [comment]
Isn't this still an issue, that should remain open? How would you implement this with sea query right now? In [the sea query docs](https://docs.rs/sea-query/0.27.1/sea_query/index/struct.IndexCreateStatement.html#method.unique), I can only find `.unique()`, which can only be used on a single column, too.

**Reply by izolyomi** [reply]
I've also just encountered this issue. Is there any conclusion on a proper solution? I'm not aware of any way to express multicolumn unique contraints.

**Reply by tyt2y3** [reply]
This is not true. You can create indices over multiple columns in SeaQuery

**Comment by tyt2y3** [comment]
https://github.com/SeaQL/sea-query/commit/6d5b3c038403890a30ce6d87f3b8c9d476da1afd
